### PR TITLE
Add integration tests related to lifecycle support to API Products

### PR DESCRIPTION
**Pull request by vithu30**
Fixes https://github.com/wso2/product-apim/issues/12015

### CODE REVIEW

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/apiproduct/lifecycle/APIProductLifecycleTest.java [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+
+        this.userMode = userMode;
+    }
+
+    @DataProvider
+    public static Object[][] userModeDataProvider() {
+
+        return new Object[][]{
+                new Object[]{TestUserMode.SUPER_TENANT_ADMIN},
+                new Object[]{TestUserMode.TENANT_ADMIN},
```
Comment (mushthaq33) [inline modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/apiproduct/lifecycle/APIProductLifecycleTest.java:81]: Shall we add the other user modes as well?

File: modules/integration/tests-integration/tests-backend/src/test/resources/artifacts/AM/configFiles/approveWorkflow/workflow-extensions.xml [hunk 2]
```diff
@@ -47,4 +47,9 @@
         <Property name="stateList">Created:Publish,Published:Block</Property>
     </APIStateChange>
 
+    <!--APIProductStateChange executor="org.wso2.carbon.apimgt.impl.workflow.APIProductStateChangeSimpleWorkflowExecutor" /-->
```
Comment (mushthaq33) [inline modules/integration/tests-integration/tests-backend/src/test/resources/artifacts/AM/configFiles/approveWorkflow/workflow-extensions.xml:50]: Shall we remove this?

Comment (vithu30) [reply modules/integration/tests-integration/tests-backend/src/test/resources/artifacts/AM/configFiles/approveWorkflow/workflow-extensions.xml:50]: This is the default executor for API Products. We have the default lifecycle executors commented out in this way for other flows too.

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/apiproduct/lifecycle/APIProductLifecycleTest.java (hunk 1 above)
Comment (vithu30) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/apiproduct/lifecycle/APIProductLifecycleTest.java:81]: I will test for other user modes and send it in a different PR

**Review by 1akshitha — APPROVED** [review]

**Review by mushthaq33 — APPROVED** [review]

File: modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/apiproduct/lifecycle/APIProductLifecycleTest.java (hunk 1 above)
Comment (vithu30) [reply modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/apiproduct/lifecycle/APIProductLifecycleTest.java:81]: Addressed in https://github.com/wso2/product-apim/pull/12116

**Pull request merged**